### ISSUE TITLE
Canon EOS 5DS capture-image-and-download different results after successive captures

### ISSUE THREAD

**Issue by joelcomp1** [gphoto/gphoto2#199]
Typically we use  --capture-image-and-download --force-overwrite to take pictures on our Canon EOS Mark III we just switched to the Canon 5DS and are seeing odd anomalies with this method.  Sometimes it works fine, other times it fails with:

ERROR: Could not capture image. ERROR: Could not capture.

So I started looking at different capture methods if i do it via:
 gphoto2 --port usb:004,014 --filename a.cr2 --trigger-capture --wait-event-and-download="CAPTURECOMPLETE" --force-overwrite  

A "good" capture outputs:
UNKNOWN PTP Property d1c3 changed
UNKNOWN Button 4
UNKNOWN OLCInfo exposure indicator 1,2,5.6 (00000000)
UNKNOWN OLCInfo event mask=41
UNKNOWN Camera Status 1
UNKNOWN PTP Property d11b changed
UNKNOWN PTP Property d1d3 changed
UNKNOWN PTP Property d1c3 changed
UNKNOWN Button 1
UNKNOWN OLCInfo exposure indicator 1,0,0.0 (00000000)
UNKNOWN OLCInfo event mask=41
Saving file as ca58518p0001.cr2
UNKNOWN PTP Property d11b changed
Saving file as ca58518p0001.cr2
CAPTURECOMPLETE
event found, stopping wait!

and a "Bad" Capture outputs (usually after 2-3 times in a row this occurs):
UNKNOWN PTP Property d1c3 changed
UNKNOWN Button 3
UNKNOWN OLCInfo exposure indicator 1,1,0.0 (00000000)
UNKNOWN OLCInfo event mask=241
UNKNOWN OLCInfo exposure indicator 1,2,5.9 (00000000)
UNKNOWN OLCInfo event mask=40
UNKNOWN PTP Property d1c3 changed
UNKNOWN Button 1
UNKNOWN OLCInfo exposure indicator 1,0,0.0 (00000000)
UNKNOWN OLCInfo event mask=241

The Button 3 vs Button 4 thing is odd to me, seems like its doing a different type of button press on the camera, and we confirmed via the LCD read out on top that it is doing that different press type.

so I then moved to:
gphoto2 --port usb:004,017 --filename a.cr2 --set-config eosremoterelease=2 --wait-event=30s --set-config eosremoterelease=4 --wait-event-and-download="CAPTURECOMPLETE" --force-overwrite

This seems to work every time so far...still trying to see if I can make it fail...

**Name the camera**
As mentioned Canon Mark III works fine, 5DS seems to have the issues.

**libgphoto2 and gphoto2 version**
gphoto2         2.5.20         gcc, popt(m), exif, no cdk, no aa, jpeg, no readline
libgphoto2      2.5.22         all camlibs, gcc, ltdl, EXIF

**Comment by msmeissn** [comment]
can you captuzre a debugoutput of the failing one?

It might not be achieving focus, or press-release is too close together

**Comment by joelcomp1** [comment]
I don't think it should be focus because we have camera in manual mode so it shouldn't be trying to do anything with focus.  I was thinking probably the latter.
Trying to get you a debug log file but now I can't get it to fail....

**Comment by msmeissn** [comment]
I did some adjustments in current libgphoto2 GIT, we detected focus finished too early on some EOS.

Might help her

**Comment by joelcomp1** [comment]
ok great thanks!
